scripts/delete-kubefed.sh

```diff
@@ -25,15 +25,15 @@ source "$(dirname "${BASH_SOURCE}")/util.sh"
 
 function delete-helm-deployment() {
   # Clean kubefed resources
-  ${KCD} -n "${NS}" FederatedTypeConfig --all
-  if [[ ! "${NAMESPACED}" || "${DELETE_CLUSTER_RESOURCE}" ]]; then
-    ${KCD} crd $(kubectl get crd | grep -E 'kubefed.io' | awk '{print $1}')
-  fi
+  # ${KCD} -n "${NS}" FederatedTypeConfig --all
```

Comment:
If this is no longer necessary, please feel free to remove. However, do we not still need: `kubectl -n kube-federation-system delete FederatedTypeConfig --all`?

Reply:
This deletes the user created federated types. This tasks hangs sometimes in my test cluster. Maybe this disturbes also the CI pipeline.

Reply:
regarding migration plan: Helm 3 is out for a while (nearly one year). It's a bit late for phasing out v2 ;)
I would prefer the hard break. Users can further install older versions in v2. But depends on the maintainer.

Reply:
totally agree with you there about helm3. @jimmidyson what are your thoughts on this?

Reply:
I would be :+1: for that, but we would need to discuss at sig-multicluster meeting to get agreement.

Can we handle upgrades from older version deployed with existing helm2-based chart to newer versions with helm3-based chart? This would be great if we can.

Reply:
This is indepenent of the chart. There is a [Helm 2to3 migration guide](https://helm.sh/blog/migrate-from-helm-v2-to-helm-v3/) where Helm v2 installations can migrate to Helm v3

Reply:
Ah that makes sense! Thank you. We should definitely add this to the docs for the release where we migrate to helm 3 though.

Reply:
@jimmidyson I've added some notes for migration: https://github.com/kubernetes-sigs/kubefed/pull/1260/commits/dd82b5232d1c7f66981d941e702f335e9fe8a3a1
Green flag from [Travis](https://travis-ci.org/github/kubernetes-sigs/kubefed/builds/715557400)

Reply:
Are we bumping the chart.yaml `apiVersion` to `v2` ? We should if we intend to using this for helm3 only.

Reply:
@alejandroEsc good point, https://github.com/kubernetes-sigs/kubefed/pull/1260/commits/a891471d139903bb5e14f5d7d90890ae5cabd232